frontend/src/static/js/v_summary.js

```diff
@@ -0,0 +1,278 @@
+    var date = new Date(datestr);
+    var day = date.getUTCDay();
+    var datems = date.getTime();
+    if(roundDown){
+        datems -= day*DAY;
+    }else{
+        datems += (7-day)*DAY;
+    }
+
+    return getDateStr(datems);
+}
+
+var vSummary = {
+    props: ["repos"],
+    template: $("v_summary").innerHTML,
+    data() {
+        return {
+            filtered: [],
+            rampScale: 0.1,
+            filterSearch: "",
+            filterSort: "totalCommits",
+            filterSortReverse: false,
+            filterGroupRepos: true,
+            filterGroupWeek: false,
+            filterSinceDate: "",
+            filterUntilDate: "",
+            filterHash: ""
+        };
+    },
+    watch:{
+        repos() { this.getFiltered(); },
+        filterSort() { this.getFiltered(); },
+        filterSortReverse() { this.getFiltered(); },
+        filterGroupRepos() { this.getFiltered(); },
+        filterGroupWeek() { this.getFiltered(); },
+        filterSinceDate() { this.getFiltered(); },
+        filterUntilDate() { this.getFiltered(); }
+    },
+    computed: {
+        sliceCount() { return this.filtered[0][0].commits.length; },
+        sliceWidth() { return 100/this.sliceCount; },
+        avgCommitSize() {
+            var totalCommits=0, totalCount=0;
+            for(var repo of this.filtered){
+                for(var user of repo){
+                    for(var slice of user.commits){
+                        if(slice.insertions===0){ continue; }
+                        totalCount += 1;
+                        totalCommits += slice.insertions;
+                    }
+                }
+            }
+            return totalCommits/totalCount;
+        },
+        avgContributionSize() {
+            var totalLines=0, totalCount=0;
+            for(var repo of this.filtered){
+                for(var user of repo){
+                    if(user.totalCommits===0){ continue; }
+                    totalCount += 1;
+                    totalLines += user.totalCommits;
+                }
+            }
+            return totalLines/totalCount;
+        }
+    },
+    methods: {
+        // view funcs
+        getWidth(slice) {
+            if(slice.insertions===0){ return 0; }
+
+            var size = this.sliceWidth;
+            size *= slice.insertions/this.avgCommitSize;
+            return Math.max(size*this.rampScale, 0.5);
+        },
+        getSliceTitle(slice) {
+            return "contribution on " + slice.fromDate +
+                ": " + slice.insertions + " lines";
+        },
+        getSliceLink(user, slice) {
+            var REPOS = window.REPOS;
+
+            return "http://github.com/" +
+              REPOS[user.repoId].organization + "/" +
+              REPOS[user.repoId].repoName + "/commits/" +
+              REPOS[user.repoId].branch + "?" +
+              "author=" + user.name + "&" +
+              "since=" + slice.fromDate + "&" +
+              "until=" + slice.toDate;
```

Comment:
do the line breaks before the `+` operator:
`+ REPOS[user.repoId].organization + "/"`...